### ISSUE TITLE
newProgressMax: spams log once per second when stdout isn't terminal

### ISSUE THREAD

**Issue by zx2c4** [restic/restic#3221]
With restic 0.11 and prior, `prune` would spit out a little progress meter with time and object count, if the output was a terminal. If the output wasn't a terminal, then instead it'd not print anything out until the end. This made for clean logs. With the new prune in git-master, I'm now seeing this in my logs:

```
Jan 07 13:37:39 coryell restic-backup.sh[349689]: + restic prune
Jan 07 13:37:42 coryell restic-backup.sh[350198]: loading indexes...
Jan 07 13:37:44 coryell restic-backup.sh[350198]: loading all snapshots...
Jan 07 13:37:44 coryell restic-backup.sh[350198]: finding data that is still in use for 1 snapshots
Jan 07 13:37:45 coryell restic-backup.sh[350198]: [0:01] 0.00%  0 / 1 snapshots
Jan 07 13:37:46 coryell restic-backup.sh[350198]: [0:02] 0.00%  0 / 1 snapshots
Jan 07 13:37:47 coryell restic-backup.sh[350198]: [0:03] 0.00%  0 / 1 snapshots
Jan 07 13:37:48 coryell restic-backup.sh[350198]: [0:04] 0.00%  0 / 1 snapshots
Jan 07 13:37:49 coryell restic-backup.sh[350198]: [0:05] 0.00%  0 / 1 snapshots
Jan 07 13:37:50 coryell restic-backup.sh[350198]: [0:06] 0.00%  0 / 1 snapshots
Jan 07 13:37:51 coryell restic-backup.sh[350198]: [0:07] 0.00%  0 / 1 snapshots
Jan 07 13:37:52 coryell restic-backup.sh[350198]: [0:08] 0.00%  0 / 1 snapshots
Jan 07 13:37:53 coryell restic-backup.sh[350198]: [0:09] 0.00%  0 / 1 snapshots
Jan 07 13:37:54 coryell restic-backup.sh[350198]: [0:10] 0.00%  0 / 1 snapshots
Jan 07 13:37:55 coryell restic-backup.sh[350198]: [0:11] 0.00%  0 / 1 snapshots
Jan 07 13:37:56 coryell restic-backup.sh[350198]: [0:12] 0.00%  0 / 1 snapshots
Jan 07 13:37:57 coryell restic-backup.sh[350198]: [0:13] 0.00%  0 / 1 snapshots
Jan 07 13:37:57 coryell restic-backup.sh[350198]: [0:13] 100.00%  1 / 1 snapshots
Jan 07 13:37:57 coryell restic-backup.sh[350198]: searching used packs...
Jan 07 13:38:00 coryell restic-backup.sh[350198]: collecting packs for deletion and repacking
Jan 07 13:38:01 coryell restic-backup.sh[350198]: [0:01] 17.30%  1000 / 5781 packs processed
Jan 07 13:38:01 coryell restic-backup.sh[350198]: [0:01] 100.00%  5781 / 5781 packs processed
Jan 07 13:38:01 coryell restic-backup.sh[350198]: to repack:            0 blobs / 0 B
Jan 07 13:38:01 coryell restic-backup.sh[350198]: this removes          0 blobs / 0 B
Jan 07 13:38:01 coryell restic-backup.sh[350198]: to delete:         5660 blobs / 214.510 MiB
Jan 07 13:38:01 coryell restic-backup.sh[350198]: total prune:       5660 blobs / 214.510 MiB
Jan 07 13:38:01 coryell restic-backup.sh[350198]: remaining:       206248 blobs / 27.008 GiB
Jan 07 13:38:01 coryell restic-backup.sh[350198]: unused size after prune: 449.753 MiB (1.63% of remaining size)
Jan 07 13:38:01 coryell restic-backup.sh[350198]: rebuilding index
Jan 07 13:38:02 coryell restic-backup.sh[350198]: [0:01] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:03 coryell restic-backup.sh[350198]: [0:02] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:04 coryell restic-backup.sh[350198]: [0:03] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:05 coryell restic-backup.sh[350198]: [0:04] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:06 coryell restic-backup.sh[350198]: [0:05] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:07 coryell restic-backup.sh[350198]: [0:06] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:08 coryell restic-backup.sh[350198]: [0:07] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:09 coryell restic-backup.sh[350198]: [0:07] 100.00%  5732 / 5732 packs processed
Jan 07 13:38:09 coryell restic-backup.sh[350198]: deleting obsolete index files
Jan 07 13:38:10 coryell restic-backup.sh[350198]: [0:01] 60.00%  3 / 5 files deleted
Jan 07 13:38:10 coryell restic-backup.sh[350198]: [0:01] 100.00%  5 / 5 files deleted
Jan 07 13:38:10 coryell restic-backup.sh[350198]: removing 49 old packs
Jan 07 13:38:11 coryell restic-backup.sh[350198]: [0:01] 8.16%  4 / 49 files deleted
Jan 07 13:38:12 coryell restic-backup.sh[350198]: [0:02] 22.45%  11 / 49 files deleted
Jan 07 13:38:13 coryell restic-backup.sh[350198]: [0:03] 40.82%  20 / 49 files deleted
Jan 07 13:38:14 coryell restic-backup.sh[350198]: [0:04] 57.14%  28 / 49 files deleted
Jan 07 13:38:15 coryell restic-backup.sh[350198]: [0:05] 75.51%  37 / 49 files deleted
Jan 07 13:38:16 coryell restic-backup.sh[350198]: [0:06] 95.92%  47 / 49 files deleted
Jan 07 13:38:17 coryell restic-backup.sh[350198]: [0:06] 100.00%  49 / 49 files deleted
Jan 07 13:38:17 coryell restic-backup.sh[350198]: done
```

It seems like that timer is being spit out when stdout isn't a terminal.

cc @aawsome

**Comment by zx2c4** [comment]
Actually, this isn't the new prune command. This is, rather, the new `newProgressMax` function. CC'ing @MichaelEischer and changing the issue title.

**Comment by MichaelEischer** [comment]
Closed as duplicate of #3194.
